google/cloud/storage/internal/async_connection_impl.h

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_ASYNC_CONNECTION_IMPL_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_INTERNAL_ASYNC_CONNECTION_IMPL_H
+
+#include "google/cloud/storage/idempotency_policy.h"
+#include "google/cloud/storage/internal/async_connection.h"
+#include "google/cloud/storage/internal/storage_stub.h"
+#include "google/cloud/storage/options.h"
+#include "google/cloud/storage/retry_policy.h"
+#include "google/cloud/completion_queue.h"
+#include "google/cloud/future.h"
+#include "google/cloud/version.h"
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace storage_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+class AsyncConnectionImpl : public AsyncConnection {
+ public:
+  explicit AsyncConnectionImpl(CompletionQueue cq,
+                               std::shared_ptr<StorageStub> stub,
+                               Options options);
+  ~AsyncConnectionImpl() override = default;
+
+  Options options() const override { return options_; }
```

Comment:
do you want a test for this? maybe that one option from `DefaultOptionGrpc()` is set?

Reply:
We check that `AuthorityOption` is set to the right value.  Is that what you meant?  I included a comment to explain that too.